app/models/GroupAccount.scala

```diff
@@ -18,9 +18,25 @@ package models
 
 import play.api.libs.json.Json
 
-case class GroupAccount(id: Long, groupId: String, companyName: String, addressId: Long, email: String, phone: String,
-                         isAgent: Boolean, agentCode: Long)
+case class GroupAccount(
+                         id: Long,
+                         groupId: String,
+                         companyName: String,
+                         addressId: Long,
+                         email: String,
+                         phone: String,
+                         isAgent: Boolean,
+                         agentCode: Option[Long]
+                       )
 
 object GroupAccount {
   implicit val format = Json.format[GroupAccount]
+
+  object AgentGroupAccount {
+    def unapply(account: GroupAccount): Option[(GroupAccount, Long)] =
+      account
```

Comment:
`PartialFunction.condOpt(account){ case a@GroupAccount(_, _, _, _, _, _, true, code) => (a, code)}` avoids mapping the agent code to a tuple if the account is not actually an agent account. but i guess it's less readable